### PR TITLE
[ADP-3385] Remove usage of deprecated `Cardano.Api.ProtocolParameters`

### PR DESCRIPTION
**Pull request by Anviking**
- Ledger-ify `mockPParams` & move to `Gen` module
- Replace `C.ProtocolParameters` with `Write.mockPParams` in wallet tests

### Comments

- I moved `mockPParams` to the `Write.Tx.Gen` module as the imports and logic should be very similar for a future generator. 
- I avoided a `Write.PParams` module as it might hinder a potential `Types.{EraIndexed, Union}` split 

### Issue Number

ADP-3385

### CODE REVIEW

File: lib/balance-tx/test/data/balanceTx/binary/testPParams/golden [hunk 1]
```diff
@@ -1 +1 @@
...
...0249f018201a000249f018201a000249f018201a000249f018201a000249f018201a000249f018201a000249f018201a00330da70101ff019f1a0003236119032c01011903e819023b00011903e8195e7104011903e818201a0001ca761928eb041959d818641959d818641959d818641959d818641959d818641959d81864186418641959d81864194c5118201a0002acfa182019b551041a000363151901ff00011a00015c3518201a000797751936f404021a0002ff941a0006ea7818dc0001011903e8196ff604021a0003bd081a00034ec5183e011a00102e0f19312a011a00032e801901a5011a0002da781903e819cf06011a00013a34182019a8f118201903e818201a00013aac0119e143041903e80a1a00030219189c011a00030219189c011a0003207c1901d9011a000330001901ff0119ccf3182019fd40182019ffd5182019581e18201940b318201a00012adf18201a0002ff941a0006ea7818dc0001011a00010f92192da7000119eabb18201a0002ff941a0006ea7818dc0001011a0002ff941a0006ea7818dc0001011a0011b22c1a0005fdde00021a000c504e197712041a001d6af61a0001425b041a00040c660004001a00014fab18201a0003236119032c010119a0de18201a00033d7618201979f41820197fb8182019a95d1820197df718201995aa18201b00000004a817c8001b00000004a817c8001a009063b91903fd0a1b00000004a817c800001b00000004a817c800ff82d81e82190241192710d81e821902d11a00989680821a00d59f801b00000002540be400821a00d59f801b00000002540be400191388189603
```
Comment (Anviking) [inline lib/balance-tx/test/data/balanceTx/binary/testPParams/golden:1]: Some params were changed to more realistic values

File: lib/balance-tx/lib/internal/Internal/Cardano/Write/Tx/Gen.hs [hunk 2]
```diff
@@ -7,28 +11,258 @@
...
+mockPParams
+    :: forall era . IsRecentEra era => PParams era
+mockPParams = case recentEra @era of
+    RecentEraBabbage -> unsafeWrap babbagePParams
+    RecentEraConway -> unsafeWrap conwayPParams
+  where
+    -- The constructor of the 'Cardano.Ledger.Core.PParams.PParams' newtype is
+    -- not exported.
+    unsafeWrap :: PParamsHKD Identity era -> PParams era
+    unsafeWrap = unsafeCoerce
```
Comment (Anviking) [inline lib/balance-tx/lib/internal/Internal/Cardano/Write/Tx/Gen.hs:105]: I prefered having the compiler enforce that we don't forget to set a field, unlike using `emptyPParams` with lens setters. Could also have used `aeson-qq`, but constructing the Haskell type gives faster feedback if something is wrong and may be more helpful for writing a generator.

File: lib/balance-tx/lib/internal/Internal/Cardano/Write/Tx/Gen.hs [hunk 3]
```diff
@@ -7,28 +11,258 @@
...
     . datumHashFromBytes
     . BS.pack <$> vectorOf 32 arbitrary
+
+--------------------------------------------------------------------------------
+-- PParams
+--------------------------------------------------------------------------------
+
+-- | We try to use similar parameters to mainnet where it matters (in particular
+-- fees, execution unit prices, and the cost model.)
+mockPParams
```
Comment (Anviking) [inline lib/balance-tx/lib/internal/Internal/Cardano/Write/Tx/Gen.hs:96]: Could be called something else, like `realisticPParams` 🤷‍♂️

**Review by HeinrichApfelmus — APPROVED** [review]
Excellent, thank you! 😊

**Pull request merged**